lib/src/main/kotlin/starknet/crypto/StarknetSignature.kt

```diff
@@ -0,0 +1,31 @@
+package starknet.crypto
+
+import org.bouncycastle.asn1.*
+import types.Felt
+import types.toFelt
+import java.io.ByteArrayOutputStream
+
+data class StarknetSignature(val r: Felt, val s: Felt) {
+    companion object {
+        @JvmStatic
+        fun fromASN1(input: ByteArray): StarknetSignature {
+            val ret = ASN1Sequence.getInstance(input)
+            val r = (ret.getObjectAt(0) as ASN1Integer).positiveValue
+            val s = (ret.getObjectAt(1) as ASN1Integer).positiveValue
```

Comment:
I think we're risking an exception here in case ret[0] or [1] cannot be cast. Maybe it'd be better to use `as?` cast and check then validate r and s?

Reply:
This means someone provided invalid input, tracing what went wrong wouldn't be a problem.

Reply:
You're probably right about tracing, but I think without explicit validation here, program will fail with something like `NullPointerException` and I don't know if that's desired behavior.

Reply:
I think we don't need it, we would throw in this case anyway. Throwing `IllegalArgumentException` provides less information than exception like `class X cannot be cast to ASN1Integer`- this way user at least knows what byte array really holds, this could be useful for debugging.